### PR TITLE
Enhance support for nil values

### PR DESCRIPTION
**Pull request by kilnerm**
This PR adds support for using nil to represent NULL for default column values.

The PR also adds test for inset, update and table creation when using nil values.

### CODE REVIEW

File: Sources/SwiftKuery/Column.swift [hunk 1]
```diff
@@ -32,6 +32,10 @@ In this example, a person `Table` class, containing three instances of the `Colu
  ```
  */
 public class Column: Field, IndexColumn {
+    // Property denoting whether default value is NULL
+    // If set to true a `nil` value for the `defaultValue` property will be interpreted as `NULL`
+    public let nullDefaultValue: Bool
```
Comment (pushkarnk) [inline Sources/SwiftKuery/Column.swift]: `nullDefaultValue` is public. If the comments above are doc comments, they'd need to start with `///`

File: Sources/SwiftKuery/Column.swift [hunk 2]
```diff
@@ -97,7 +101,8 @@ public class Column: Field, IndexColumn {
      - Parameter check: The expression to check for values inserted into the column. Defaults to nil.
      - Parameter collate: The collation rule for the column. Defaults to nil.
      */
```
Comment (pushkarnk) [inline Sources/SwiftKuery/Column.swift]: We might want to add documentation for the new parameter

File: Sources/SwiftKuery/ColumnCreator.swift [hunk 3]
```diff
@@ -26,4 +26,56 @@ public protocol ColumnCreator {
     /// - Parameter column: The column being built
     /// - Returns: A string representation of the column for the implementing database or nil if it cannot be built
     func buildColumn(for column: Column, using queryBuilder: QueryBuilder) -> String?
+
+    /// Build an appropriate representation of a passed value for the database
+    /// A default implemetation is provided that can be overriden should a
+    /// plugin require non-common behaviour
+    ///
+    /// - Parameter item: The value to convert
```
Comment (pushkarnk) [inline Sources/SwiftKuery/ColumnCreator.swift]: Parameter documentation for `queryBuilder` seems to be missing

File: Sources/SwiftKuery/ColumnCreator.swift [hunk 4]
```diff
@@ -26,4 +26,56 @@ public protocol ColumnCreator {
...
+    /// - Returns: A string representing the value that can be passed into the database
+    func packType(_ item: Any, queryBuilder: QueryBuilder) throws -> String
+
+    /// Get the dafault value for a column
```
Comment (pushkarnk) [inline Sources/SwiftKuery/ColumnCreator.swift]: typo: `default`

File: Sources/SwiftKuery/ColumnCreator.swift [hunk 5]
```diff
@@ -26,4 +26,56 @@ public protocol ColumnCreator {
...
+public extension ColumnCreator {
+
+    func getDefaultValue(for column: Column, queryBuilder: QueryBuilder) -> String? {
+        guard let defaultValue = column.defaultValue else {
+            if column.nullDefaultValue {
+                return "NULL"
+            }
+            return nil
+        }
+        do{
```
Comment (pushkarnk) [inline Sources/SwiftKuery/ColumnCreator.swift]: nit: a whitespace after `do`?

File: Sources/SwiftKuery/Insert.swift [hunk 6]
```diff
@@ -81,25 +81,35 @@ public struct Insert: Query {
...
     /// - Parameter values: An array of values (the row) to insert.
-    public init(into table: Table, values: [Any], returnID: Bool=false) {
+    public init(into table: Table, values: [Any?], returnID: Bool=false) {
         self.init(into: table, columns: nil, values: values, returnID: returnID)
     }
+    //#else
+    /// Initialize an instance of Insert.
+    ///
+    /// - Parameter into: The table to insert rows.
+    /// - Parameter values: An array of values (the row) to insert.
```
Comment (pushkarnk) [inline Sources/SwiftKuery/Insert.swift]: typo: must be `Parameter optionalValues`

Comment (pushkarnk) [reply Sources/SwiftKuery/Insert.swift]: Parameter `returnID` needs to be documented

File: Sources/SwiftKuery/Insert.swift [hunk 7]
```diff
@@ -81,25 +81,35 @@ public struct Insert: Query {
     ///
     /// - Parameter into: The table to insert rows.
     /// - Parameter values: A list of values (the row) to insert.
-    public init(into table: Table, values: Any..., returnID: Bool=false) {
+    public init(into table: Table, values: Any?..., returnID: Bool=false) {
         self.init(into: table, columns: nil, values: values, returnID: returnID)
     }
     
+   // #if swift(>=4.1.0)
```
Comment (pushkarnk) [inline Sources/SwiftKuery/Insert.swift]: Can we remove the `#if-#else-#endif` that seems to have been commented out?

**Comment by kilnerm** [comment]
I have reconsider the api changes involved here and concluded they carry too much risk of accidental breakage.

I am going to re-work the API to make this change safer.
